mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java

```diff
@@ -85,6 +87,36 @@ default void notificationShouldNotExceedRate() {
             assertThat(rateExceeded).isFalse();
         }
 
+        @Test
+        default void notificationShouldDeliverASingleEventToAllListenersAtTheSameTime() {
+            CountDownLatch countDownLatch = new CountDownLatch(1);
+            try {
+                ConcurrentLinkedQueue<String> threads = new ConcurrentLinkedQueue<>();
+                eventBus().register(event -> {
+                    threads.add(Thread.currentThread().getName());
+                    countDownLatch.await();
+                }, KEY_1);
+                eventBus().register(event -> {
+                    threads.add(Thread.currentThread().getName());
+                    countDownLatch.await();
+                }, KEY_1);
+                eventBus().register(event -> {
+                    threads.add(Thread.currentThread().getName());
+                    countDownLatch.await();
+                }, KEY_1);
+
+                eventBus().dispatch(EVENT, KEY_1).subscribeOn(Schedulers.elastic()).subscribe();
+
+
+                WAIT_CONDITION.atMost(org.awaitility.Duration.TEN_SECONDS)
+                    .untilAsserted(() -> assertThat(threads.size()).isEqualTo(3));
```

Comment:
All same comments than for GroupContract.

Reply:
ok